packages/task/src/browser/task-configurations.ts

```diff
@@ -84,30 +84,41 @@ export class TaskConfigurations implements Disposable {
         @inject(FileSystem) protected readonly fileSystem: FileSystem
     ) {
         this.toDispose.push(watcherServer);
-        this.watcherServer.onFilesChanged(async changes => {
-            try {
-                const watchedConfigFileChanges = changes.filter(change =>
-                    this.watchedConfigFileUris.some(fileUri => FileChangeEvent.isAffected([change], new URI(fileUri)))
-                ).map(relatedChange => (
-                    { uri: relatedChange.uri.toString(), type: relatedChange.type }
-                ));
-                if (watchedConfigFileChanges.length >= 0) {
-                    await this.onDidTaskFileChange(watchedConfigFileChanges);
-                    if (this.client) {
-                        this.client.taskConfigurationChanged(this.getTaskLabels());
+        this.toDispose.push(
+            this.watcherServer.onFilesChanged(async changes => {
+                try {
+                    const watchedConfigFileChanges = changes.filter(change =>
+                        this.watchedConfigFileUris.some(fileUri => FileChangeEvent.isAffected([change], new URI(fileUri)))
+                    ).map(relatedChange => (
+                        { uri: relatedChange.uri.toString(), type: relatedChange.type }
+                    ));
+                    if (watchedConfigFileChanges.length >= 0) {
+                        await this.onDidTaskFileChange(watchedConfigFileChanges);
+                        if (this.client) {
+                            this.client.taskConfigurationChanged(this.getTaskLabels());
+                        }
                     }
+                } catch (err) {
+                    console.error(err);
                 }
-            } catch (err) {
-                console.error(err);
-            }
-        });
-
+            })
+        );
         this.toDispose.push(Disposable.create(() => {
             this.tasksMap.clear();
+            this.taskCustomizationMap.clear();
+            this.watchersMap.clear();
             this.client = undefined;
         }));
     }
 
+    @postConstruct()
+    protected init(): void {
+        this.toDispose.push(this.taskDefinitionRegistry);
+        this.toDispose.push(
+            this.taskDefinitionRegistry.onDidChangeTaskDefinition(() => this.reorgnizeTasks())
```

Comment:
Should not we call `this.reorgnizeTasks` in `init` as well? In the case if task definitions were registered before?